src/librustc/dep_graph/dep_node.rs

```diff
@@ -171,16 +173,20 @@ macro_rules! define_dep_nodes {
                 }
             }
 
-            #[inline]
-            pub fn is_input(&self) -> bool {
+            #[inline(always)]
+            pub fn is_input_inlined(&self) -> bool {
```

Comment:
`is_anon` and `is_eval_always` has uses where inlining them might not be beneficial. A regular `#[inline]` does not suffice since these are large functions and LLVM won't inline them unless you tell it to. I use the `*_inlined` variants for calls when LLVM can optimize it all away.

I want to turn these properties into associated constants anyway, but that is blocked on https://github.com/rust-lang/rust/pull/56462.

Reply:
I think switching to `debug_assertions` will remove most of the hot calls to those functions. Since they might be replaced soon anyway, I'd prefer not to split the functions in order to avoid churn. You can just mark them as `#[inline(always)]` if performance is bad otherwise.